LiteFarmOrg/LiteFarm
# Added unassigned notification, did small refactor

This PR addresses https://lite-farm.atlassian.net/browse/LF-2463

To test:
1. Create a task and assign it someone (you should be on an account with farm management permissions)
2. Unassign that task
3. You should get a notification on all management accounts for that farm that the task has been unassigned
4. The notification should not appear for any worker accounts
5. Because I did a little refactoring in the code that sends notifications based on tasks, test some other use cases of task notifications and ensure they are still sending properly
